EIPS/eip-5299.md

```diff
@@ -0,0 +1,168 @@
+---
+eip: 5299
+title: Layer3 scaling pattern
+description: A primitive to facilitate eternal upgradeability of smart contracts, by making storage slots dynamic.
+author: Vinay Kumar (@v9hstk), Pratik Agarwal (@Prtk418)
+discussions-to: https://ethereum-magicians.org/t/eip-9000-layer3-scaling-standard/9039
+status: Draft
+type: Standards Track
+category: ERC
+created: 2021-07-18
+requires: 2535
+---
+
+## Abstract
+Upgrading a storage layout is required in case of a complex project if it is re-architected. The EIP introduces dynamic storage slots, which can be upgraded without deploying a new contract.
+
+## Motivation
+With the existing smart contract upgradable design patterns available, while it’s possible to upgrade the code implementation, it is not possible to completely upgrade the storage structure, which requires a re-deployment of the contracts.
+
+From the end-user perspective, it is annoying to migrate from an old version of an app to a new version, especially if it requires deploying multiple transactions.
+
+
+## Specification
```

Comment:
Thank you for the EIP! Please consider complete the spec section, otherwise it's hard for readers to understand what kind of practice / interface this EIP proposes other people to follow.